azurerm/internal/services/mssql/tests/resource_arm_mssql_virtual_machine_test.go

```diff
@@ -0,0 +1,594 @@
+					resource.TestCheckResourceAttr(data.ResourceName, "server_configuration.0.sql_connectivity_type", "PRIVATE"),
+					resource.TestCheckResourceAttr(data.ResourceName, "server_configuration.0.sql_connectivity_port", "1433"),
+				),
+			},
+			data.ImportStep("server_configuration.0.sql_connectivity_update_password", "server_configuration.0.sql_connectivity_update_username"),
+			{
+				Config: testAccAzureRMMsSqlVirtualMachine_withServerConfigUpdated(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMsSqlVirtualMachineExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "server_configuration.0.is_r_services_enabled", "true"),
+					resource.TestCheckResourceAttr(data.ResourceName, "server_configuration.0.sql_connectivity_type", "PUBLIC"),
+					resource.TestCheckResourceAttr(data.ResourceName, "server_configuration.0.sql_connectivity_port", "1533"),
+				),
+			},
+			data.ImportStep("server_configuration.0.sql_connectivity_update_password", "server_configuration.0.sql_connectivity_update_username"),
+		},
+	})
+}
+
+func testCheckAzureRMMsSqlVirtualMachineExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Sql Virtual Machine not found: %s", resourceName)
+		}
+
+		id, err := parse.MssqlVmID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		client := acceptance.AzureProvider.Meta().(*clients.Client).MSSQL.SQLVirtualMachinesClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.Name, ""); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Sql Virtual Machine (Sql Virtual Machine Name %q / Resource Group %q) does not exist", id.Name, id.ResourceGroup)
+			}
+			return fmt.Errorf("Bad: Get on sqlVirtualMachinesClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMMsSqlVirtualMachineDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).MSSQL.SQLVirtualMachinesClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_mssql_virtual_machine" {
+			continue
+		}
+
+		id, err := parse.MssqlVmID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.Name, ""); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on sqlVirtualMachinesClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMVirtualMachine_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-mssqlvm-%[1]d"
```

Comment:
could we just use the service package name abbreviation 
```suggestion
  name     = "acctestRG-mssql-%[1]d"
```